### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by hobbes1069** [OpenImageIO/oiio#3208]
**Describe the bug**
In the spirit of the discussion during the ASWF meeting, making option things optional :)

Even if tests are disabled (`OIIO_BUILD_TESTS=FALSE`) CMake still tries to clone oiio-tests.

**To Reproduce**
1. Add -DOIIO_BUILD_TEST=FALSE option to CMake
2. Run CMake
3. Then THIS happens:
```
-- Cloning oiio-images from https://github.com/OpenImageIO/oiio-images.git
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Git (missing: GIT_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindGit.cmake:128 (find_package_handle_standard_args)
  src/cmake/testing.cmake:346 (find_package)
  src/cmake/testing.cmake:357 (oiio_get_test_data)
  CMakeLists.txt:289 (oiio_setup_test_data)
```

**Expected behavior**
All the test related CMake commands should be wrapped in a conditional not to execute if testing is not enabled/used.

**Platform information:**
 - OIIO branch/version: 2.3.10
 - OS: Fedora Linux
 - C++ compiler: 11.2.1
 - Any non-default build flags when you build OIIO:
N/A

Probably update the existing conditional for PROJECT_IS TOP_LEVEL?

**Comment by lgritz** [comment]
You can disable this with `-DOIIO_DOWNLOAD_MISSING_TESTDATA=OFF`

So there is an interesting subtlety with `OIIO_BUILD_TESTS`: Currently, it literally means "should we **build** the executable unit tests" (like simd_test, imagebuf_test, etc). It doesn't mean "should we run tests". (Should it?)

Tests are only run when `PROJECT_IS_TOP_LEVEL` (i.e., not for subprojects), and you're probably also correct that this disable the downloading of any test data.


OIIO_BUILD_TESTS currently literally means "

**Comment by lgritz** [comment]
I see now that `PROJECT_IS_TOP_LEVEL` already does also prevent (for subprojects) the downloads from happening.

**Comment by lgritz** [comment]
So I think we are left with this:

* Does knowing about `DOIIO_DOWNLOAD_MISSING_TESTDATA` completely solve this issue for you?

* Do you think that `OIIO_BUILD_TESTS`, when disabled, should disable *all* testing and preparation thereof, not merely disable the building of certain additional programs that will be later used for some tests? (Those seem like separate questions to me -- (a) do you build these test programs, which can be useful even when not running the testsuite, versus (b) do you do various things to set up for later running the testsuite?)

**Comment by hobbes1069** [comment]
Well here's one of the things I don't like about CMake, you can bury options anywhere and they're not visible until you run cmake at least once, which for package building is inconvenient since it's not interactive. But now I know about the option I can use it.

Right now it fails for local builds because I don't install git in the `chroot`, but I think my basic premise here is that the configuring should never fail for anything that's optional.

I can think of a few options:

1. Test for git and just skip downloading if it's not available, a warning about it would be fine.
2. Add an option for `SKIP_TESTING` or whatever verbiage you prefer (positive or negative incantation) that implies `OIIO_DOWNLOAD_TESTDATA` and `OIIO_BUILD_TESTS` is `FALSE`.

Or some other option we can think of. The key, I believe, is to make things "just work" for people who are building manually so they don't need to do a bunch of extra steps (like downloading the test data set) and that it (mostly) "just works" for packaging as well, i.e., not failing on optional stuff.

**Comment by lgritz** [comment]
OH, I see now. At the core of the problem is that the whole thing goes kablooey if git can't be found. That should be an easy fix, let me try...

**Comment by lgritz** [comment]
Proposed fix #3212
